Source/WebInspectorUI/UserInterface/Models/DOMNodeStyles.js

```diff
@@ -358,7 +358,7 @@ WI.DOMNodeStyles = class DOMNodeStyles extends WI.Object
         return this._pendingRefreshTask;
     }
 
-    addRule(selector, text, styleSheetId)
+    addRule(selector, text, styleSheetId, addedRuleCallback)
```

Comment:
It seems like we should either make this method `async` and have it return the new new rule ID

Reply:
I don't think that'll work.

The entire chain of operations within `DOMNodeStyles.addRule()` ends with a call to `DOMNodeStyles.refresh()`. That will trigger a `WI.DOMNodeStyles.Event.Refreshed` event in response to which `WI.SpreadsheetRulesStyleDetailsPanel` will do layout. The `styleID` is needed in that layout step in order to mark the correct rule as editable.

Awaiting the full set of operations will lead to a race condition.

Here we're intentionally calling the `addedRuleCalback` synchronously as soon as the rule is created, before any other method has a chance to call `DOMNodeStyles.refresh()` and trigger the layout above.

Reply:
Oh, neat! That'll work. Updated the patch.